### PR TITLE
[TEMPURA-2060] Contribute to Backpack library Badge icon with Painter

### PR DESCRIPTION
**Pull request by anton-sobolev-skyscanner**
Copied functional from BpkButton to BpkBadge to add support for custom Badge icons. Compose only

| Before | Now |
|--------|-----|
| Screenshot_20241128_095339 | Screenshot_20241128_104440|

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [x] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+private val BpkBadgeType.borderColor: Color
+    @Composable
+    get() = when (this) {
+        BpkBadgeType.Outline -> BpkTheme.colors.textOnDark
+        else -> Color.Transparent
+    }
+
+private val BadgeShape = RoundedCornerShape(BpkBorderRadius.Xs)
+
+private fun Modifier.defaultIconSize(size: BpkIconSize): Modifier =
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt:154]: i don't think this is needed? badges should only ever have one icon size, so we don't need the `BpkIconSize` prop here - they're always small

File: app/src/main/java/net/skyscanner/backpack/demo/compose/BadgeStory.kt [hunk 2]
```diff
@@ -95,6 +97,14 @@ private fun BadgeRow(
                 else -> BpkIcon.TickCircle
             },
         )
+
+        BpkBadge(
```
Comment (marianeum) [inline app/src/main/java/net/skyscanner/backpack/demo/compose/BadgeStory.kt:101]: please regenerate the docs screenshots as well

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+internal fun BpkBadgeImpl(
+    text: String,
+    modifier: Modifier = Modifier,
+    type: BpkBadgeType = BpkBadgeType.Normal,
+    icon: @Composable (RowScope.() -> Unit)?,
+) {
+    Row(
+        modifier = modifier
+            .semantics(mergeDescendants = true) { }
+            .border(1.dp, type.borderColor, BadgeShape)
```
Comment (marianeum) [inline backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt:57]: not new, but while we're here: `BpkBorderSize.Sm`

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt (hunk 1 above)
Comment (anton-sobolev-skyscanner) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt:154]: removed

File: backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt (hunk 3 above)
Comment (anton-sobolev-skyscanner) [reply backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/badge/internal/BpkBadgeImpl.kt:57]: fixed

File: app/src/main/java/net/skyscanner/backpack/demo/compose/BadgeStory.kt (hunk 2 above)
Comment (anton-sobolev-skyscanner) [reply app/src/main/java/net/skyscanner/backpack/demo/compose/BadgeStory.kt:101]: updated

**Review by marianeum — APPROVED** [review]

**Pull request merged**